packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+            bottom: 0,
+            height: 0,
+            left: 0,
+            right: 0,
+            top: 0,
+            width: 0,
+            x: 0,
+            y: 0,
+          };
+          return defaultDOMRect;
+        }
+        return this.sliderElement.current.getBoundingClientRect();
+      },
+      getTabIndex: (): number => this.props.tabIndex || 0,
+      // React handles events differently:
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      registerThumbContainerInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        if (evtType === this.transitionendEvtName) {
+          this.onThumbEnd = handler as any;
+        }
+      },
+      deregisterThumbContainerInteractionHandler: (
+        evtType: EventType
+      ): void => {
+        if (evtType === this.transitionendEvtName) {
+          this.onThumbEnd = undefined;
+        }
+      },
+      registerBodyInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        document.body.addEventListener(evtType, handler);
+      },
+      deregisterBodyInteractionHandler: (
+        evtType: EventType,
+        handler: SpecificEventListener<EventType>
+      ): void => {
+        document.body.removeEventListener(evtType, handler);
+      },
+      registerResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.addEventListener('resize', handler);
+      },
+      deregisterResizeHandler: (
+        handler: SpecificEventListener<'resize'>
+      ): void => {
+        window.removeEventListener('resize', handler);
+      },
+      notifyInput: (): void => {
+        const callback = this.props.onValueInput;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      notifyChange: (): void => {
+        const callback = this.props.onValueChange;
+        if (typeof callback === 'function') {
+          callback(this.foundation.getValue());
+        }
+      },
+      setThumbContainerStyleProperty: (prop: string, value: string): void =>
+        this.setStyleToElement(prop, value, 'thumbContainerStyleProperty'),
+      setTrackStyleProperty: (prop: string, value: string): void =>
+        this.setStyleToElement(prop, value, 'trackStyleProperty'),
+      setMarkerValue: (value: number): void =>
+        this.setState({markerValue: value}),
+      appendTrackMarkers: (numMarkers: number): void => {
+        this.setState((prevState) => {
+          let {markerStyles} = prevState;
+          // Shallow copy of array:
+          markerStyles = markerStyles.slice();
+          for (let i = 0; i < numMarkers; i++) {
+            markerStyles.push({});
+          }
+          return {markerStyles};
+        });
+      },
+      removeTrackMarkers: (): void => {
```

Comment:
remove this void...its unnecessary

Reply:
Should I remove the other void return specifications?